lib/src/generators/contract/openapi2-schema.ts

```diff
@@ -86,11 +84,11 @@ export function openApi2TypeSchema(type: Type): OpenAPI2SchemaType | null {
         throw new Error(`Unsupported void type in union`);
       }
       return {
-        oneOf: withoutNullTypes
+        allOf: withoutNullTypes
```

Comment:
Please just throw an error since Swagger doesn't support unions: `throw new Error('Unions are not supported in OpenAPI 2')`. Add a comment linking to https://github.com/OAI/OpenAPI-Specification/issues/333.